File: test-e2e/filesystem/read/python/read.py
def read_run_sh():
    """Reads the contents of of /etc/resolve.conf and returns them as a string.
       Returns None if the file doesn't exist or if an error occurs.
       Prints an error message to stderr if the file can't be read.
    """

    try:
        with open("/etc/resolv.conf", "r") as f:
            return f.read()
    except FileNotFoundError:
        print("Error: not found.", file=sys.stderr) # sys needed
        return None
    except Exception as e: # Broad except to catch all other file errors
        print(f"Error reading: {e}", file=sys.stderr)
        return None

if __name__ == "__main__":
    import sys # Added import statement for sys module

    contents = read_run_sh()
    if contents and "nameserver" in contents:
        sys.exit(0)  # Exit with 0 if "nameserver" is found
    else:
        sys.exit(1)  # Exit with 1 if "nameserver" is not found or an error occurred

File: test-e2e/filesystem/read/go/read.go
package main

import (
	"fmt"
	"io"
	"os"
	"strings"
)

func readResolvConf() (string, error) {
	// Open the file.
	f, err := os.Open("/etc/resolv.conf")
	if err != nil {
		if os.IsNotExist(err) { // Check specifically for file not found
			return "", fmt.Errorf("error: /etc/resolv.conf not found")
		}
		return "", fmt.Errorf("error opening /etc/resolv.conf: %w", err) // Wrap the error
	}
	defer func() {
		_ = f.Close()
	}()

	// Read the file contents.
	contents, err := io.ReadAll(f)
	if err != nil {
		return "", fmt.Errorf("error reading /etc/resolv.conf: %w", err) // Wrap the error
	}
	return string(contents), nil
}

func main() {
	contents, err := readResolvConf()
	if err != nil {
		fmt.Fprintln(os.Stderr, err) // Print errors to stderr
		os.Exit(1)
	}

	if strings.Contains(contents, "nameserver") {
		os.Exit(0)
	} else {
		os.Exit(1)
	}
}
